### PR TITLE
[fix]:[CI-5714]: Fix backend compatibility issue for docker runner

### PR DESCRIPTION
**Pull request by happyhebaby**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38184)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38184)

**Comment by harshjain12** [comment]
We can not revert it as changing type is not allowed without adding new field.

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/vm/CIVmExecuteStepTaskParams.java [hunk 1]
```diff
@@ -42,7 +42,8 @@ public class CIVmExecuteStepTaskParams implements CIExecuteStepTaskParams, Execu
   @NotNull private String logKey;
   @NotNull private String workingDir;
   private Map<String, String> volToMountPath;
-  @NotNull @Getter private InfraInfo infraInfo;
+  //@Getter private InfraInfo infraInfo;
+  @Getter private CIInitializeTaskParams.Type infraInfo;
```
Comment (aman-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/vm/CIVmExecuteStepTaskParams.java:46]: Wouldn't this also introduce backward compatibility issues? Is this class already added in Kryo?

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/vm/CIVmInitializeTaskParams.java [hunk 2]
```diff
@@ -66,7 +65,8 @@
   private String stageRuntimeId;
   @Expression(ALLOW_SECRETS) private List<VmServiceDependency> serviceDependencies;
   @Builder.Default private static final Type type = Type.VM;
-  @NotNull @Getter private InfraInfo infraInfo;
+  //@Getter private InfraInfo infraInfo;
+  @Getter private CIInitializeTaskParams.Type infraInfo;
```
Comment (aman-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/vm/CIVmInitializeTaskParams.java:69]: Same as above comment.

**Review by happyhebaby — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38184)* status: 32 file(s) to review. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/vm/CIVmExecuteStepTaskParams.java` line 46 at r2](https://reviewable.io/reviews/harness/harness-core/38184#-NDCERo9Qfdz9BIe8sZG-r2-46:-NDKFmpf3qYc2-FU7QAt:b-eyrgq2) ([raw file](https://github.com/harness/harness-core/blob/83f293f6b68b32f95d43da024c40dd7ea4f11fcb/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/vm/CIVmExecuteStepTaskParams.java#L46)):*
<details><summary><i>Previously, aman-harness (Aman Singh) wrote…</i></summary><blockquote>

Wouldn't this also introduce backward compatibility issues? Is this class already added in Kryo?
</blockquote></details>

The type seems to already get registered in this line in `DelegateTasksBeansKryoRegister`:
```
kryo.register(CIInitializeTaskParams.Type.class, 19397);
```
So I guess we are good. I already tested locally with exact setting that I reproed the error (old delegate, other services uses new version), with this fix, it didn’t generate any error.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/vm/CIVmInitializeTaskParams.java` line 69 at r2](https://reviewable.io/reviews/harness/harness-core/38184#-NDCERo9Qfdz9BIe8sZH-r2-69:-NDKG7WLB6xQ45R43okL:bs4q9op) ([raw file](https://github.com/harness/harness-core/blob/83f293f6b68b32f95d43da024c40dd7ea4f11fcb/950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/ci/vm/CIVmInitializeTaskParams.java#L69)):*
<details><summary><i>Previously, aman-harness (Aman Singh) wrote…</i></summary><blockquote>

Same as above comment.
</blockquote></details>

Ditto.

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 21 of 22 files at r1, 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38184#-:-NDKfJ5tBdklCbvt-DYc:bnfp4nl)* status: :shipit: complete! ✓

**Review by happyhebaby — COMMENTED** [review]
TFTR

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38184#-:-NDMjHHACDcMROUtbEj9:bb73zsl)* status: :shipit: complete! ✓

**Review by harshjain12 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38184)* status: :shipit: complete! ✓

**Review by harshjain12 — APPROVED** [review]
> Reviewed 21 of 22 files at r1, 10 of 11 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38184)* status: :shipit: complete! ✓

**Comment by dgarg-harness** [comment]
codebase hash mismatch is approved by Aman. Proceeding with force merge

**Pull request merged**